test/es6/toStringTag.js

```diff
@@ -5,299 +5,221 @@
 
-            class MyClass {
-                constructor() { }
+                assert.areEqual("[object " + builtInName + "]", Object.prototype.toString.call(i), builtInName + " defaults to internal slot based tag when @@toStringTag override is not string coercible");
             }
 
-            var m = new MyClass();
-            var m2 = new MyClass();
-            assert.areEqual("[object Object]", Object.prototype.toString.call(m), "toString should have tag Object");
-
-            m2[Symbol.toStringTag] = "MyClassInstance";
-            assert.areEqual("[object MyClassInstance]", Object.prototype.toString.call(m2), "toString should have tag MyClassInstance");
-            assert.areEqual("[object Object]", Object.prototype.toString.call(m), "toString should have tag Object m2 only changed instance tag");
-
-            MyClass.prototype[Symbol.toStringTag] = "MyClass";
-            assert.areEqual("[object MyClassInstance]", Object.prototype.toString.call(m2), "toString should have tag MyClassInstance because MyClass is higher in the prototype chain");
-            assert.areEqual("[object MyClass]", Object.prototype.toString.call(m), "toString should have tag MyClass because prototype tag changed");
+            for (var { tag } of [...explicitTagExtensibleBuiltIns, ...explicitTagTypedArrayBuiltIns, ...internalSlotBuiltIns]) {
+                testBuiltIn(tag);
+            }
        }
     },
-    */
     {
-       name: "throws test case",
-       body: function () {
-            var a = {};
-            a[Symbol.toStringTag] = "aObject";
-            assert.areEqual("aObject", a[Symbol.toStringTag], "check Tag is aObject");
-
-            assert.areEqual("[object aObject]", Object.prototype.toString.call(a), "check toString has tag aObject");
-            Object.defineProperty(a, Symbol.toStringTag, {get : function() { throw 10;}});
-
+        name: "@@toStringTag lookup does not use HasProperty",
+        body: function() {
+             var proxy = new Proxy({ [Symbol.toStringTag] : "Proxied" }, {
+                // Previous versions of the spec called HasProperty
+                has: function (target, key) {
+                    assert.fail("Object.prototype.toString should not call HasProperty");
+                }
+            });
+            assert.areEqual("[object Proxied]", Object.prototype.toString.call(proxy), "Proxied @@toStringTag passes through value without calling HasProperty");
+        }
+    },
+    {
+        name: "@@toStringTag properties with side effects",
+        body: function () {
+            var tagThrows = { get [Symbol.toStringTag]() { throw new Error(); } };
+            var threw = false;
             try {
-                a[Symbol.toStringTag];
-            } catch(e) {
-                assert.areEqual("[object ???]", Object.prototype.toString.call(a), "JS exceptions should have tag ???");
+                Object.prototype.toString.call(instance);

```

Comment:
nit: why not change this test to assert.throws()  instead of having the try..catch and a variable?